### PR TITLE
kubectl not waiting infinitely

### PR DESCRIPTION
**Pull request by comm-rat**

### CODE REVIEW

File: vindaloo/vindaloo.py [hunk 1]
```diff
@@ -239,6 +240,23 @@ def k8s_deploy(self) -> None:
                     self._out('Waiting for rollout {} to finish'.format(deployment_name))
                     self.cmd(["kubectl", "rollout", "status", "deployment", deployment_name])
 
+            for yaml_conf in self.config_module.K8S_OBJECTS.get('job', []):
+                if isinstance(yaml_conf, JsonSerializable):
+                    job_name = yaml_conf.name
+                if job_name:
```
Comment (zdenekbauer) [inline vindaloo/vindaloo.py:246]: Radši bych měl mezeru nad tímhle `if` blokem a tu pod dal pryč

**Review by zdenekbauer — APPROVED** [review]

File: vindaloo/vindaloo.py [hunk 2]
```diff
@@ -239,6 +240,23 @@ def k8s_deploy(self) -> None:
...
+                    while not succeeded and not failed:
+                        time.sleep(60)  # one minute wait for not messing up kube
+                        succeeded = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.succeeded}"]).stdout
+                        failed = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.failed}"]).stdout
+                        if succeeded:
+                            self._out("Job {} ended successfully.".format(job_name))
+                        if failed:
+                            self._out("Job {} failed.".format(job_name))
+                            exit(1)
+
```
Comment (norraxx) [inline vindaloo/vindaloo.py:259]: Once, in very important moment, I've met situation, when the pod was in loop on failing readiness probe.

File: vindaloo/vindaloo.py [hunk 3]
```diff
@@ -239,6 +240,23 @@ def k8s_deploy(self) -> None:
...
+                    succeeded, failed = None, None
+                    while not succeeded and not failed:
+                        time.sleep(60)  # one minute wait for not messing up kube
+                        succeeded = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.succeeded}"]).stdout
+                        failed = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.failed}"]).stdout
+                        if succeeded:
+                            self._out("Job {} ended successfully.".format(job_name))
+                        if failed:
+                            self._out("Job {} failed.".format(job_name))
+                            exit(1)
```
Comment (norraxx) [inline vindaloo/vindaloo.py:258]: Is it the problem of the vindaloo, that kube job failed, to exit with 1?

File: vindaloo/vindaloo.py [hunk 4]
```diff
@@ -238,6 +239,25 @@ def k8s_deploy(self) -> None:
...
+                    while not succeeded and not failed:
+                        time.sleep(60)  # one minute wait for not messing up kube
+                        succeeded = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.succeeded}"]).stdout
+                        failed = self.cmd(["kubectl", "get", "job", job_name, "-o", "jsonpath={.status.failed}"]).stdout
+                        if succeeded:
+                            self._out("Job {} ended successfully.".format(job_name))
+                        if failed:
+                            failed = True
+                            self._out("Job {} failed.".format(job_name))
+            if should_exit:
```
Comment (norraxx) [inline vindaloo/vindaloo.py:259]: where do you set this `should_exit`?

**Comment by zdenekbauer** [comment]
@dundee ❤️ 👋

**Comment by dundee** [comment]
Looks good to me :) Thanks for your contribution!

**Pull request merged**